internal/gatewayapi/route.go

```diff
@@ -190,6 +191,33 @@ func (t *Translator) processHTTPRouteRules(httpRoute *HTTPRouteContext, parentRe
 						}
 						route.Destination.Settings = append(route.Destination.Settings, ds)
 						route.BackendWeights.Valid += backendWeight
+						route.BackendTLS = func() *ir.TLSBundle {
+							targetBackend := GetTargetBackendReference(backendRef.BackendRef, NamespaceDerefOr(NamespacePtr(httpRoute.Namespace), "default"))
+							err, tlsBundle := getBackendTLSBundle(resources.BackendTLSPolicies, resources.ConfigMaps, resources.Secrets, targetBackend)
```

Comment:
this code should live inside processDestination
https://github.com/envoyproxy/gateway/blob/cf46fbe776918ad19444e26d637ffcc79676ca23/internal/gatewayapi/route.go#L1067

Also suggest moving this logic into a seperate method like `processBackendTLSPolicy`

Reply:
moved